cmd/postgres-operator/main.go

```diff
@@ -86,8 +86,27 @@ func initManager() (runtime.Options, error) {
 		options.LeaderElectionNamespace = os.Getenv("PGO_NAMESPACE")
 	}
 
-	if namespace := os.Getenv("PGO_TARGET_NAMESPACE"); len(namespace) > 0 {
-		options.Cache.DefaultNamespaces = map[string]runtime.CacheConfig{namespace: {}}
+	// Check PGO_TARGET_NAMESPACE for backwards compatibility with
+	// "singlenamespace" installations
+	singlenamespace := os.Getenv("PGO_TARGET_NAMESPACE")
+
+	// Check PGO_TARGET_NAMESPACES for non-cluster-wide, multi-namespace
+	// installations
+	multinamespace := os.Getenv("PGO_TARGET_NAMESPACES")
+
+	// Initialize DefaultNamespaces if any target namespaces are set
+	if len(singlenamespace) > 0 || len(multinamespace) > 0 {
+		options.Cache.DefaultNamespaces = map[string]runtime.CacheConfig{}
+	}
+
+	if len(singlenamespace) > 0 {
+		options.Cache.DefaultNamespaces[singlenamespace] = runtime.CacheConfig{}
+	}
+
+	if len(multinamespace) > 0 {
+		for _, namespace := range strings.Split(multinamespace, ",") {
```

Comment:
🤔 🤔 If we ignore/remove whitespace, then `len(string) > 0` might not be the right check. Another way to arrange this could be to build up a `map[string]runtime.CacheConfig` then assign `options.Cache.DefaultNamespaces` when `len(map) > 0`.

Reply:
Not sure I follow your last message... If we strip whitespace and all that is left is an empty string, then no namespaces were provided.

I like the `FieldsFunc` approach for splitting up the string.